src/goto-instrument/contracts/contracts.cpp

```diff
@@ -174,6 +174,17 @@ void code_contractst::check_apply_loop_contracts(
     return invariant_copy;
   };
 
+  // Process "loop_entry" history variables
+
+  std::map<exprt, exprt> parameter2history;
+  goto_programt history;
```

Comment:
As per our offline discussion, may be just reuse `havoc_code` instead of defining this?

Reply:
Not addressed yet.